qiskit/providers/ibmq/ibmqfactory.py

```diff
@@ -473,7 +480,11 @@ def _initialize_providers(self, credentials: Credentials) -> None:
             try:
                 provider = AccountProvider(provider_credentials, self)
                 self._providers[provider_credentials.unique_id()] = provider
+                if lazy:
+                    self.__lazy_loaded = True
+                    return
```

Comment:
So this would return after loading the first provider. But any subsequent call to `providers()`, including `get_provider()`, will still load all providers (and duplicate the `user_hubs()` call).

`load_account()` is slow mostly for people with multiple providers because it loads all providers (by this I mean getting the backend information for each provider). If I do
```
default_provider = IBMQ.load_account()
provider = IBMQ.get_provider('foo')
```
only providers `default_provider` and  `foo` need to be loaded.

Reply:
Fixed!